### PR TITLE
MGMT-8769- Fail olm test on operator failure.

### PR DESCRIPTION
**Pull request by eliorerz**
Use cluster fixture in test_olm_operator instead of outdated fixture get_cluster.
Raise error on new_cluster_configuration on missing ClusterConfig attribute

/cc @osherdp

### CODE REVIEW

File: src/assisted_test_infra/test_infra/helper_classes/cluster.py [hunk 1]
```diff
@@ -738,7 +738,7 @@ def wait_for_operators_to_finish(self, timeout=consts.CLUSTER_INSTALLATION_TIMEO
             cluster_id=self.id,
             operators_count=len(operators_utils.filter_operators_by_type(operators, OperatorType.OLM)),
             operator_types=[OperatorType.OLM],
-            statuses=[consts.OperatorStatus.AVAILABLE, consts.OperatorStatus.FAILED],
+            statuses=statuses,
```
Comment (lalon4) [inline src/assisted_test_infra/test_infra/helper_classes/cluster.py:741]: why this change is needed? we have fall_on_error_status if you expect failure

Comment (eliorerz) [reply src/assisted_test_infra/test_infra/helper_classes/cluster.py:741]: But we don't use those here. It always wait for failures regardless of `fall_on_error_status` value.
Also, If we don't mind if the operator is success or failure we can now use the `fall_on_error_status`

Comment (lalon4) [reply src/assisted_test_infra/test_infra/helper_classes/cluster.py:741]: why not using it? it does considered in _are_operators_in_status

**Pull request merged**